packages/sanity/src/tasks/src/tasks/context/enabled/TasksEnabledProvider.tsx

```diff
@@ -12,22 +12,33 @@ interface TaksEnabledProviderProps {
  * @internal
  */
 export function TasksEnabledProvider({children}: TaksEnabledProviderProps) {
-  const {enabled, isLoading} = useFeatureEnabled('studioTasks')
+  const {enabled, isLoading} = useFeatureEnabled('sanityTasks')
+
+  // Staging flag - keeping here until the new flag is added to staging.
+  const {enabled: stagingIsEnabled, isLoading: stagingIsLoading} = useFeatureEnabled('studioTasks')
+
   const isWorkspaceEnabled = useWorkspace().tasks?.enabled
 
   const value: TasksEnabledContextValue = useMemo(() => {
-    if (!enabled || !isWorkspaceEnabled || isLoading) {
+    if (!isWorkspaceEnabled || isLoading || stagingIsLoading) {
       return {
         enabled: false,
         mode: null,
       }
     }
+    // The staging check will be removed when the sanityTasks flag is added to staging.
+    if (stagingIsEnabled || enabled) {
+      return {
+        enabled: true,
+        mode: 'default',
+      }
+    }
 
     return {
-      enabled: true,
-      mode: 'default',
+      enabled: false,
+      mode: null,
```

Comment:
This in the future will return mode `upsell` that's why it's on a different return